lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -383,9 +384,6 @@ def _on_peer_relation_created(self, event: RelationCreatedEvent):
                 "Adding units during an upgrade is not supported. The charm may be in a broken, unrecoverable state"
             )
 
-        # Store the "Admin" certificate, key and CA on the disk of the new unit
-        self.tls.store_admin_tls_secrets_if_applies()
```

Comment:
Same as below:

This caused trouble with the locking mechanism: If the operator tries to connect with Opensearch using the "new" admin certificate with the already "new" CA, it can't connect and then at some point the leader just used the peer databag lock to restart upon CA-renewal, independent if another unit may already be restarting. I could fix this by not storing the admin-certs during an ongoing CA-renewal, but this can only be controlled by deferring the `CertificateAvailableEvent`. Storing admin-certs on other occasions (like `peer-relation changed/created) was destroying this deferral-approach, that's why I removed it from these handlers and decided to only store when really sure it's fine.